datalake_api/querier.py

```diff
@@ -177,29 +176,35 @@ def __init__(self, table_name, dynamodb=None):
         self.dynamodb = dynamodb
 
     def query_by_work_id(self, work_id, what, where=None, cursor=None):
-        kwargs = self._prepare_work_id_kwargs(work_id, what, cursor)
+        kwargs = self._prepare_work_id_kwargs(work_id, what)
         if where is not None:
             self._add_range_key_condition(kwargs, where)
+        if cursor is not None:
+            self._add_cursor_conditions(kwargs, cursor)
+
         response = self._table.query(**kwargs)
         cursor = self._cursor_for_work_id_query(response)
         return QueryResults(response['Items'], cursor)
 
-    def _prepare_work_id_kwargs(self, work_id, what, cursor):
+    def _prepare_work_id_kwargs(self, work_id, what):
         i = work_id + ':' + what
-        kwargs = dict(
-            IndexName='work-id-index',
-            KeyConditionExpression=Key('work_id_index_key').eq(i),
-            Limit=MAX_RESULTS,
-        )
-        if cursor is not None:
-            self._add_cursor_conditions(kwargs, cursor)
-        return kwargs
+        return {
+            'IndexName': 'work-id-index',
+            'ExpressionAttributeNames': {
+                '#n0': 'work_id_index_key',
+            },
+            'ExpressionAttributeValues': {
+                ':v0': i
+            },
+            'KeyConditionExpression': '#n0 = :v0',
+            'Limit': MAX_RESULTS,
+        }
 
     def _add_range_key_condition(self, kwargs, where):
-        condition = kwargs['KeyConditionExpression']
-        where_condition = Key('range_key').begins_with(where + ':')
-        new_condition = And(condition, where_condition)
-        kwargs['KeyConditionExpression'] = new_condition
+        q = '(#n0 = :v0 AND begins_with(#n1, :v1))'
+        kwargs['ExpressionAttributeNames']['#n1'] = 'range_key'
+        kwargs['ExpressionAttributeValues'][':v1'] = where + ':'
+        kwargs['KeyConditionExpression'] = q
```

Comment:
Consolidate or move closer to L204 to emphasize the decoupling with lines 205/206?